### ISSUE TITLE
Icons need a way to have an accessible label, and alternatively, a way to be marked as presentational when no label is present

### ISSUE THREAD

**Issue by nicklemmon** [SparkPost/matchbox#212]
#193 is related (this issue will need to be resolved first, actually).

Essentially, the icon component as it currently exists does not have a way to have a screen reader only label. As a result, if a user encounters a meaningful icon with a screen reader, it will not be interpreted properly.

Similarly, icon markup should be updated when an icon instance is purely presentational vs. informational.

## Possible Implementation

At a previous company, this was the route we took, and it's an approach outlined by [CSS Tricks](https://css-tricks.com/accessible-svgs/#article-header-id-9 as well.

1. Add a new prop such as `title` or `label` that can be optionally passed in to an icon instance. Let's go with `label` for consistency's sake
1. When the `label` value is present, the `<svg>` tag needs the following updates:
  * Render the aria attribute `aria-label={ label }`
  * Additionally, `role="img"` should render as well
  * And `aria-hidden="false"`
1. When the `label` value is *not* present:
  * the `<svg>` tag should have the attribute `aria-hidden="true"`
